### ISSUE TITLE
Addon Resets Start Time and Adds Reminder to Task Created in TickTick

### ISSUE THREAD

**Issue by zarb1n** [thesamim/TickTickSync#98]
Hey thesamim,

I'm returning with a bug report that, while not critical, is incredibly frustrating.

### Bug Description

After creating a task in TickTick, setting both start and end times (I primarily use the calendar view), and syncing it via the addon, the task in TickTick loses its start time and instead acquires a reminder set to "On time".

### To Reproduce

1. Create a task in TickTick and specify both start and end times.
2. Wait for the addon to sync the task.
3. Observe that the start time has been removed, and a reminder has been added to the task.

### Expected Behavior

The addon should not remove the start time of the task or add a reminder to it during synchronization.

### Software Versions

I'm utilizing the **Windows desktop** applications with the following software versions:
- Obsidian: v1.5.11
- TickTick: v5.2.0.0
- TickTickSync: 1.0.22

Your assistance in resolving this issue would be greatly appreciated. Thank you!

**Comment by thesamim** [comment]
What is end time? That's not something I'm seeing in Tasks.

**Comment by zarb1n** [comment]
Yeah, you're correct, Tasks doesn't support time as far as I know. However, in TickTick, users have the option to specify both start and end times for tasks -- it helps to manage tasks in calendar view.

For example, I've included screenshots below (before sync and after):

![Before sync](https://github.com/thesamim/TickTickSync/assets/60680932/939bbd06-5c97-48f2-917e-00a77c34726d)
![After sync](https://github.com/thesamim/TickTickSync/assets/60680932/71a8f604-0971-4297-8bb9-fef8dd2090a4)

**Comment by thesamim** [comment]
Ah. End time is also due date.  Will work on that.

Will also look into recurrences issue. Note, though: I don't think that Tasks and TickTick recurrence are compatible. We'll see....

**Comment by zarb1n** [comment]
Apologies for any confusion caused. As a non-native English speaker, I might make mistakes like this.

I believe syncing recurrences between Tasks and TickTick might not be necessary since it seems to be a consequence of the addon resetting all "time" properties, including recurrences.

**Comment by thesamim** [comment]
> Apologies for any confusion caused. As a non-native English speaker, I might make mistakes like this.
> 
Please don't apologize.  Communication through any text based medium is going to be challenging. Your English is very clear. 

My understanding not so much. :) 

> I believe syncing recurrences between Tasks and TickTick might not be necessary since it seems to be a consequence of the addon resetting all "time" properties, including recurrences.

There are two sets of recurrence indicators. There is the recurrence data set from TickTick. And there's the recurrence data set from Tasks.

Which one is disappearing for you?

**Comment by zarb1n** [comment]
I use the TickTick feature for recurring tasks (*so I can see them in a calendar view*).

**Comment by thesamim** [comment]
> I use the TickTick feature for recurring tasks (_so I can see them in a calendar view_).

Got it. Will look into it. Thanks for the update.

**Comment by Adanufgail** [comment]
Ah, this explains what I'm seeing I think. I'm frequently getting bombarded with EVERY task for today sending an alert at 9 AM despite me never setting up a reminder. Also, I'm often seeing recurring tasks that have synced with Obsidian AND have been postponed are no longer creating new tasks when completed in TickTick (I use sync as an archive).

**Comment by thesamim** [comment]
@zarb1n , @Adanufgail : After updating to 1.0.40 (and looking at https://github.com/thesamim/TickTickSync/wiki/Notable-Changes#changes-from-1036-to-1040----new-datetime-handling) please let me know if you're still having issues.

**Comment by Adanufgail** [comment]
Updated and restarted. I'll let you know if I see it happen again!

**Comment by thesamim** [comment]
@zarb1n , @Adanufgail : I believe this issue is fixed. If you're still having issues, please re-open and give me the details.
